scrapy_crawlera/middleware.py

```diff
@@ -211,3 +215,19 @@ def _is_crawlera_header(self, header_name):
             return False
         header_name = header_name.decode('utf-8').lower()
         return header_name.startswith(self.header_prefix.lower())
+
+    def _set_crawlera_default_headers(self, request):
+        for header, value in self._headers:
+            if value is None:
+                continue
+            request.headers.setdefault(header, value)
+        lower_case_headers = [
+            header.decode('utf-8').lower() for header in request.headers
+        ]
+        if all(h.lower() in lower_case_headers for h in self.conflicting_headers):
+            logging.warn(
```

Comment:
Does it makes sense to log this just once per run?

Reply:
Good question, I see two possibilities:

1. Log on first occurrence;
2. Log everytime, but add the url to identify where this is happening to the user.

I lean more towards 2, but can be too much, don't know.

Reply:
I'd go with 2 as well, as it captures one-off issues from different places in the spider, as the user can pass conflicting headers in a per-request basis.

Reply:
I added the url and went with option 2. @baitxo if you have strong objections let me know.